protocol/lib/eth/util.go

```diff
@@ -21,27 +21,16 @@ const (
 	MaxAddrLen = 32
 )
 
-// ABI Singleton and mutex.
-var bridgeEventAbi *ethabi.ABI
-var once sync.Once
-
-// getBridgeEventAbi returns the ABI (application binary interface) for the Bridge contract.
-func GetBridgeEventAbi() *ethabi.ABI {
-	// Initialize the singleton in a thread-safe way.
-	once.Do(func() {
-		bridgeAbi, err := ethabi.JSON(strings.NewReader(constants.BridgeEventABI))
+// GetBridgeEventAbi returns the ABI (application binary interface) for the Bridge contract.
+func GetBridgeEventAbi() ethabi.ABI {
+	result := sync.OnceValue[ethabi.ABI](func() ethabi.ABI {
```

Comment:
You want to define `result` outside of this method as a global variable otherwise `sync.OnceValue()` will be created each time you invoke this method.